extensions/amp-ooyala-player/0.1/amp-ooyala-player.js

```diff
@@ -0,0 +1,139 @@
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {user} from '../../../src/log';
+import {removeElement} from '../../../src/dom';
+
+class AmpOoyalaPlayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://player.ooyala.com', opt_onLayout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    const embedCode = user().assert(
+        this.element.getAttribute('data-embedcode'),
+        'The data-embedcode attribute is required for <amp-ooyala-player> %s',
+        this.element);
+    const pCode = user().assert(
+        this.element.getAttribute('data-pcode'),
+        'The data-pcode attribute is required for <amp-ooyala-player> %s',
+        this.element);
+    const playerId = user().assert(
+        this.element.getAttribute('data-playerid'),
+        'The data-playerid attribute is required for <amp-ooyala-player> %s',
+        this.element);
+
+    let src = 'https://player.ooyala.com/iframe.html?platform=html5-priority';
+    const playerVersion = this.element.getAttribute('data-playerversion') || '';
+    if (playerVersion.toLowerCase() == 'v4') {
+      src = 'https://player.ooyala.com/static/v4/stable/latest/skin-plugin/amp_iframe.html?pcode='
+        + encodeURIComponent(pCode);
+      const configUrl = this.element.getAttribute('data-config');
+      if (configUrl) {
+        src += '&options[skin.config]=' + encodeURIComponent(configUrl);
+      }
+    }
+
+    src += '&ec=' + encodeURIComponent(embedCode) +
+      '&pbid=' + encodeURIComponent(playerId);
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = src;
+    this.applyFillContent(iframe);
+
+    this.element.appendChild(iframe);
+    return this.loadPromise(iframe);
+  }
+
+  /** @override */
+  play(unusedIsAutoplay) {
+    this.iframe_.contentWindow.postMessage('play', '*');
+  }
+
+  /** @override */
+  pause() {
+    this.iframe_.contentWindow.postMessage('pause', '*');
+  }
+
+  /** @override */
+  mute() {
+    this.iframe_.contentWindow.postMessage('mute', '*');
+  }
+
+  /** @override */
+  unmute() {
+    this.iframe_.contentWindow.postMessage('unmute', '*');
+  }
+
+  /** @override */
+  pauseCallback() {
+    if (this.iframe_) {
+      this.pause();
+    }
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    return true;
+  }
+
+  /** @override */
+  unlayoutOnPause() {
+    return false;
+  }
+
+  /** @override */
+  supportsPlatform() {
+    return true;
+  }
+
+  /** @override */
+  isInViewport() {
```

Comment:
I have rebased this branch with master (git rebase master) and there were some merge conflicts in gulpfile.js and test-example-validation.js. I resolved conflicts by re-applying just our changes to master versions of these files, checked them in, but somehow a lot of old code in these files is still seen in this pull request. Then I deleted local copy of amphtml and re-checked out amp-ooyala branch and I now I am seeing that this branch is gone!  
Fortunately, I had all copies of files in my editor and I created a new amp-oolaya branch (it allowed me to do git checkout -b amp-ooyala) and restored all our code and modifications to other files.  But now I can't push this branch as I don't have permissions:
git push -u origin amp-ooyala
remote: Permission to ampproject/amphtml.git denied to mityaha.
fatal: unable to access 'https://github.com/ampproject/amphtml.git/': The requested URL returned error: 403

What did I do wrong? What was I supposed to rebase our branch with?  Why did our amp-ooyala branch go away? What should I do now?  Please help!

Also, on each commit I was performing, I was receiving email from googlebot, "Thanks for your pull request. It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA)."  I did sign agreement, I did reply with "I signed it!" message, but I keep receiving these responses from googlebot.  How can this problem be fixed?